core/src/main/kotlin/com/strumenta/kolasu/model/Naming.kt

```diff
@@ -65,3 +67,27 @@ fun <N> ReferenceByName<N>.tryToResolve(possibleValue: N?): Boolean where N : Po
         true
     }
 }
+
+interface Symbol : PossiblyNamed
+
+data class Scope(
+    val symbols: MutableMap<String, MutableList<Symbol>> = mutableMapOf(),
```

Comment:
Maybe we could make symbols private and expose methods to add/read from this table, to ensure it is always used correctly. I.e., we do not want to insert symbols under the wrong key but the key to be always the same as the name of the symbol. Also, we want only to insert symbols with names

Reply:
👍